services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3809,27 +3799,20 @@ private void setRmdChunkingEnabled(String clusterName, String storeName, boolean
   void setIncrementalPushEnabled(String clusterName, String storeName, boolean incrementalPushEnabled) {
     storeMetadataUpdate(clusterName, storeName, store -> {
       VeniceControllerClusterConfig config = getHelixVeniceClusterResources(clusterName).getConfig();
-      if (incrementalPushEnabled) {
+      if (incrementalPushEnabled || store.isHybrid()) {
         // Enabling incremental push
+        store.setNativeReplicationEnabled(config.isNativeReplicationEnabledAsDefaultForHybrid());
+        store.setNativeReplicationSourceFabric(config.getNativeReplicationSourceFabricAsDefaultForHybrid());
         store.setActiveActiveReplicationEnabled(
```

Comment:
Should we move this out? Like we are not determining this by inc push.. also for the below branch. (This is a bit confusing, please feel free to correct me again)

Reply:
I would prefer keeping it this way until we move all hybrid stores to AA & incpush.